### ISSUE TITLE
Cant fetch thumbnails

### ISSUE THREAD

**Issue by NowakJR** [thibaudgg/video_info#206]
Hey, so lately I've gotten into a problem with fetching thumbnails from vimeo, every single thumbnail in a project was vertical bars in colours of a rainbow.  It turned out vimeo changed their url format, and didnt bother to tell anyone(thread: https://community.roku.com/t5/Roku-Developer-Program/Vimeo-Thumbnail-is-not-showing-in-any-channel/td-p/721844)
So to get to the real thumbnails which are here, under pictures/sizes/link I had to monkeypatch this gem.

"pictures": {
    "uri": "/videos/604925161/pictures/1241392221",
    "active": true,
    "type": "custom",
    "sizes": [
      {
        "width": 100,
        "height": 75,
        "link": "https://i.vimeocdn.com/video/1241392221-cd13a462737b76c8d54c15bf8d38b15575bc04d18031d4653c97d0b2c8eb6db3-d_100x75?r=pad",

Are you planning on fixing this issue? 
Cheers.

**Comment by pozelli** [comment]
Hey
How did you do the monkeypatch?
Thanks

**Comment by fakenine** [comment]
Hey @NowakJR. I'd also be interested for the monkey patch while we wait for a fix, thanks :)

**Comment by NowakJR** [comment]
Hi guys here is how I did the monkey patch: In file where i am fetching thumbnail URL i added these lines of code in Base class
```
VideoInfo::Providers::VimeoAPI.module_eval do

            def thumbnail_id

              @thumbnail_id ||= _parse_picture_id(_video['pictures']['sizes'].first['link'])

            end

            def _parse_picture_id(string)

              a, b, c = string.split("/").last.split("-")

              d = c.split("_").first

              [a, b, d].join("-")

            end

          end
```
I know variables names are kinda bad but I left them like that for a while until i have time to change them.

**Comment by fakenine** [comment]
@NowakJR Thank you! I will test this

**Comment by pozelli** [comment]
Yeah, it works! Thanks!

I created the file: config/initializers/video_info.rb with the following code:
`VideoInfo::Providers::VimeoAPI.module_eval do
  def thumbnail_id
    @thumbnail_id ||= _parse_picture_id(_video['pictures']['sizes'].first['link'])
  end
  def _parse_picture_id(uri)
    a, b, c = uri.split("/").last.split("-")
    d = c.split("_").first
    [a, b, d].join("-")
  end
end`

**Comment by asgerb** [comment]
I've found that the patch fails on certain videos, here's a cleaned up version which works with all the videos I've tested:

```ruby
VideoInfo::Providers::VimeoAPI.module_eval do
  THUMBNAIL_LINK_REGEX = /.*\/(\d+\-[^_]+)/

  def thumbnail_id
    @thumbnail_id ||= _parse_picture_id(_video.dig("pictures", "sizes").first["link"])
  end

  private

  def _parse_picture_id(uri)
    uri.match(THUMBNAIL_LINK_REGEX)[1]
  end
end

```

**Comment by NowakJR** [comment]
Awesome man! This also looks much better. Thank you very much for contributing @asgerb

**Comment by ruanltbg** [comment]
Thank you @asgerb  and @NowakJR . That worked fine. 

Hey @joaocv3 , any idea when we could have this merged in?

**Comment by ruanltbg** [comment]
In some cases the first split has only two results, I had to modify the code to accept this case. 

This is my code:
```
VideoInfo::Providers::VimeoAPI.module_eval do
  def thumbnail_id
    @thumbnail_id ||= _parse_picture_id(_video['pictures']['sizes'].first['link'])
  end

  private

  def _parse_picture_id(uri)
    # separate the uri in numeral parts, some cases the third part is
    # not available.
    first, second, third = uri.split("/").last.split("-")
    if third.nil?
      third = second
      second = nil
    end
    last = third.split("_").first
    [first, second, last].compact.join("-")
  end
end
```
This works with these two types of videos:
- https://vimeo.com/245409458 (URI: "https://i.vimeocdn.com/video/670087024-8045c3b78cc946670e843fa0d1beb4e803c9a438347c6ed3b709e2022cacd292-d_100x75?r=pad")
-https://vimeo.com/623543586  (URI: "https://i.vimeocdn.com/video/1267692865-430e40ccadb8e88a0381466d872b3e621282a0f3121fe4ad3_100x75?r=pad")

**Comment by joaocv3** [comment]
Hi guys, thanks for the inputs!

I couldnt find much time to work on it for the last weeks

just released v3.0.2 🚀
